src/server/scripts/Spells/spell_paladin.cpp

```diff
@@ -919,6 +920,44 @@ class spell_pal_hand_of_salvation : public SpellScriptLoader
     }
 };
 
+// -20335 - Heart of the Crusader
+class spell_pal_heart_of_the_crusader : public SpellScriptLoader
+{
+    public:
+        spell_pal_heart_of_the_crusader() : SpellScriptLoader("spell_pal_heart_of_the_crusader") { }
+
+        class spell_pal_heart_of_the_crusader_AuraScript : public AuraScript
+        {
+            PrepareAuraScript(spell_pal_heart_of_the_crusader_AuraScript);
+
+            bool Validate(SpellInfo const* /*spellInfo*/) override
+            {
+                return ValidateSpellInfo({ SPELL_PALADIN_HEART_OF_THE_CRUSADER_EFF_R1 });
+            }
+
+            void HandleProc(AuraEffect const* aurEff, ProcEventInfo& eventInfo)
+            {
+                PreventDefaultAction();
+
+                uint32 spellId = sSpellMgr->GetSpellWithRank(SPELL_PALADIN_HEART_OF_THE_CRUSADER_EFF_R1, GetSpellInfo()->GetRank());
+                if (auto actor = eventInfo.GetActor())
+                {
+                    actor->CastSpell(eventInfo.GetProcTarget(), spellId, aurEff);
+                }
+            }
+
+            void Register() override
+            {
+                OnEffectProc += AuraEffectProcFn(spell_pal_heart_of_the_crusader_AuraScript::HandleProc, EFFECT_0, SPELL_AURA_DUMMY);
```

Comment:
```suggestion
                OnEffectProc += AuraEffectProcFn(spell_pal_heart_of_the_crusader_AuraScript::HandleProc, EFFECT_0, SPELL_EFFECT_APPLY_AURA);
```

Because Effect 0 in DBC use it, to prevent the error reported above
```
ID - 20335 Heart of the Crusader (Rank 1)
=================================================
Description: In addition to the normal effect, your Judgement spells will also increase the critical strike chance of all attacks made against that target by an additional $s1%.
Category = 0, SpellIconID = 237, activeIconID = 0, SpellVisual = (0,0), SpellPriority = 0
Family SPELLFAMILY_PALADIN, flag [0] 0x00000000 [1] 0x00000000 [2] 0x00000000

SpellSchoolMask = 1 (SPELL_SCHOOL_MASK_NORMAL)
DamageClass = 0 (SPELL_DAMAGE_CLASS_NONE)
PreventionType = 0 (SPELL_PREVENTION_TYPE_NONE)
=================================================
Attributes: 0x000001D0 (SPELL_ATTR0_ABILITY, SPELL_ATTR0_PASSIVE, SPELL_ATTR0_HIDDEN_CLIENTSIDE, SPELL_ATTR0_HIDE_IN_COMBAT_LOG)
=================================================
Skill (Id 184) "Retribution"
    ReqSkillValue 1, Forward Spell = 0, MinMaxValue (0, 0), CharacterPoints (0, 0)Spell Level = 0, base 0, max 0, maxTarget 0

Category = 0
DispelType = 0 (DISPEL_NONE)
Mechanic = 0 (MECHANIC_NONE)
SpellRange: (Id 1) "Self Only":
    MinRange = 0, MinRangeFriendly = 0
    MaxRange = 0, MaxRangeFriendly = 0

CastingTime (Id 1) = 0,00

Interrupt Flags: 0x00000000, AuraIF 0x00000000, ChannelIF 0x00000000
Chance = 0, charges - 0
=================================================
Effect 0: Id 6 (SPELL_EFFECT_APPLY_AURA)
BasePoints = 1 Bonus = 1,00
Targets (1, 0) (TARGET_UNIT_CASTER, NO_TARGET)
Aura Id 4 (SPELL_AURA_DUMMY), value = 1, misc = 12 (12), miscB = 0, periodic = 0
SpellClassMask = 20000000 00000000 00000000
	- 21183 - Heart of the Crusader (Rank 1)
	- 26135 - Battlegear of Eternal Justice
	- 54498 - Heart of the Crusader (Rank 2)
	- 54499 - Heart of the Crusader (Rank 3)
	- 61840 - Righteous Vengeance

Effect 1:  NO EFFECT

Effect 2:  NO EFFECT
```